routers/web/repo/editor.go

```diff
@@ -685,7 +685,11 @@ func UploadFilePost(ctx *context.Context) {
 
 	message := strings.TrimSpace(form.CommitSummary)
 	if len(message) == 0 {
-		message = ctx.Tr("repo.editor.upload_files_to_dir", form.TreePath)
+		dir := form.TreePath
+		if dir == "" {
+			dir = "/"
+		}
+		message = ctx.Tr("repo.editor.upload_files_to_dir", dir)
```

Comment:
I still don't think that localizing commit messages is a good idea.
Commits should be written in English and **only** English.
If someone wants to write them in their native language, go for it, but I don't think we should encourage an antipattern by default.
But I know this is offtopic for this PR, so my approval isn't influenced by that.

Reply:
That could be another PR.